rosplan_planning_system/src/ProblemGeneration/PDDLProblemGenerator.cpp

```diff
@@ -271,28 +291,38 @@ namespace KCL_rosplan {
 				rosplan_knowledge_msgs::KnowledgeItem attr = currentGoalSrv.response.attributes[i];
 				if(attr.knowledge_type == rosplan_knowledge_msgs::KnowledgeItem::FACT) {
 
-                    switch(attr.is_negative){
-
-                        case true:
-                        {
-                            pFile << "    (not(" + attr.attribute_name;
-                            for(size_t j=0; j<attr.values.size(); j++) {
-                                pFile << " " << attr.values[j].value;
-                            }
-                            pFile << ")";
-                        }
-                        break;
-
-                        case false:
-                        {
-                            pFile << "    (" + attr.attribute_name;
-                            for(size_t j=0; j<attr.values.size(); j++) {
-                                pFile << " " << attr.values[j].value;
-                            }
-                        }
-                        break;
+                    if(attr.is_negative){pFile << "    (not("+ attr.attribute_name;}
+                    else {pFile << "    (" + attr.attribute_name;}
+
+                    for(size_t j=0; j<attr.values.size(); j++) {
+                        pFile << " " << attr.values[j].value;
                     }
+
+                    if(attr.is_negative){pFile << ")";}
+
                     pFile << ")" << std::endl;
+
+
+				}
+				if(attr.knowledge_type == rosplan_knowledge_msgs::KnowledgeItem::INEQUALITY) {
+
+                    // do we need comparison negation? remove if not
```

Comment:
I think we do need negation. It is strange, but I think it is possible to put a PDDL goal:
(not (> x y))
We'll need to also check the problem visitor to check for such a thing.

Reply:
Just looked above to the problem parser -- checking for negation and setting it in the inequality item is already done, so this is all fine.

Reply:
Ok